File: patterns/creational/singleton/Singleton.py
class Singleton:
    def __new__(cls):
        if cls.instance is None:
            cls.instance = object.__new__(cls)
            return cls.instance
        else:
            raise Exception("Error, singletons can only be instanciated once.")
    @classmethod
    def get(cls):
        if cls.instance is None:
            cls.instance = object.__new__(cls)
        return cls.instance

# class monManager(Singleton):
#     instance = None
#     def __new__(cls):
#         return Singleton.__new__(cls)
#     def aff(self, x):
#         print(x)
#     @staticmethod
#     def aff2(x):
#         monManager.get().aff(x)
# 
# #a=monManager()
# print(monManager.get())
# monManager.get().aff("inst")
# monManager.aff2("stat")
# a=monManager()
# print(a)

File: patterns/creational/singleton/Singleton.php
<?php
    abstract class Singleton {
        function __construct() {
            if (self::isInstancied()) { throw new Exception(get_called_class()." is a Singleton, so you can instanciate it only once."); }
            eval(get_called_class()."::\$instance=\$this;");
        }
        protected static function isInstancied() {
            $b=null;
            eval("\$b=".get_called_class()."::\$instance!=null;");
            return $b;
        }
        public static function get() { 
            $inst=null;
            if (!self::isInstancied()) eval("new ".get_called_class()."();");
            eval("\$inst=".get_called_class()."::\$instance;");
            return $inst;
        }
    }
?>